### PR TITLE
feat(headless): support DictionaryFieldContext

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/KIT-354

### CODE REVIEW

**Review by btaillon — APPROVED** [review]

**Review by francoislg — COMMENTED** [review]
Can you also add this in the `ProductRecommendationsAppState` (in another PR)? :smile:

File: packages/headless/src/controllers/dictionary-field-context/headless-dictionary-field-context.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+/**
+ * The `DictionaryFieldContext` controller allows specifying which [dictionary field](https://docs.coveo.com/en/2036/index-content/about-fields#dictionary-fields) keys to retrieve.
+ */
+export interface DictionaryFieldContext extends Controller {
+  /**
+   * Sets the dictionary field context for the query. This replaces any existing context with the new one.
+   *
+   *  @param ctx - The dictionary field context to set for the query.
+   */
+  set(ctx: DictionaryFieldContextPayload): void;
```
Comment (ThibodeauJF) [inline packages/headless/src/controllers/dictionary-field-context/headless-dictionary-field-context.ts:24]: dont think we are gaining much from that shortening of context to ctx, I would just leave the full word

**Comment by ThibodeauJF** [comment]
How should we allow users to set these context/dictionary context with Atomic? With a component or going through Headless?

**Review by ThibodeauJF — APPROVED** [review]

**Comment by samisayegh** [comment]
> Can you also add this in the ProductRecommendationsAppState

Sure, I'll do this in a dedicated PR,

**Comment by samisayegh** [comment]
> How should we allow users to set these context/dictionary context with Atomic?

Considering it's non-visual, I think using javascript to call functions would be the most consistent, and most flexible.

File: packages/headless/src/controllers/dictionary-field-context/headless-dictionary-field-context.ts (hunk 1 above)
Comment (samisayegh) [reply packages/headless/src/controllers/dictionary-field-context/headless-dictionary-field-context.ts:24]: adjusted, thanks

**Pull request merged**